### PR TITLE
keys: stop printing {a-/Min} if a range is {Key: "a", EndKey: nil}

### PR DESCRIPTION
**Pull request by spencerkimball**
Release note: None

### CODE REVIEW

File: pkg/keys/printer.go [hunk 1]
```diff
@@ -693,9 +693,14 @@ func MassagePrettyPrintedSpanForTest(span string, dirs []encoding.Direction) str
 // PrettyPrintRange pretty prints a compact representation of a key range. The
 // output is of the form:
 //    commonPrefix{remainingStart-remainingEnd}
+// If the end key is empty, the outut is of the form:
+//    start
 // It prints at most maxChars, truncating components as needed. See
 // TestPrettyPrintRange for some examples.
 func PrettyPrintRange(start, end roachpb.Key, maxChars int) string {
+	if len(end) == 0 {
+		return prettyPrintInternal(nil /* valDirs */, start, false /* quoteRawKeys */)
```
Comment (tschottdorf) [inline pkg/keys/printer.go]: You should still take into account `maxChars`. (Imagine `len(start) = 10000`).

Comment (spencerkimball) [reply pkg/keys/printer.go]: Done.

**Comment by RaduBerinde** [comment]
LGTM
nit: can move prettyStart before the if instead of duplicating it

**Review by tbg — APPROVED** [review]

**Comment by spencerkimball** [comment]
Done.

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/21508#-:-L3Del2muf6j7-VIl-24:b-896fix)*

**Pull request merged**